publish-kedro-viz action Initial Release (#1)
kedro-org/publish-kedro-viz: README.md

![Kedro Logo Banner - Dark](https://raw.githubusercontent.com/kedro-org/kedro/main/.github/demo-light.png#gh-light-mode-only)

<br />
<p align="center">

![Kedro-Viz Pipeline Visualisation](https://raw.githubusercontent.com/kedro-org/kedro-viz/main/.github/img/banner.png)

</p>

<p align="center">
✨ <em> Data Science Pipelines. Beautifully Designed</em> ✨
<br />
Live Demo: <a href="https://demo.kedro.org/" target="_blank">https://demo.kedro.org/</a>
</p>

<br />

## Overview

A GitHub action which deploys your Kedro-Viz static site on GitHub pages. This action helps in the automation of a deployment startegy mentioned in [platform agnostic sharing with Kedro-Viz](https://docs.kedro.org/projects/kedro-viz/en/v8.0.1/platform_agnostic_sharing_with_kedro_viz.html#static-website-hosting-platforms-such-as-github-pages)

## Prerequisites

- **GitHub Pages Setup:** Configure your repository for [GitHub Pages](https://docs.github.com/en/pages/quickstart).
- **Kedro-project dependencies:** Install all the Kedro-project dependencies before using this action in your workflow
- **Python-version:** You need to have an environment with `python>=3.9` in your workflow

> **NOTE:** While configuring your repository for GitHub Pages, you have two publishing source options. You can either choose a `branch` or `a custom GitHub Actions workflow`. If you choose a branch, the build artifacts will be uploaded to the `publish_branch` you pass as an input to the action, which defaults to `gh-pages`. If you choose a custom GitHub Actions workflow, you need to mention that in the input `publishing_source` to the action and no artifacts/branch will be created. Please find more information on configuring a publishing source for github pages site in the [official docs](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site).

Reviewer: ```suggestion
**NOTE:** While configuring your repository for GitHub Pages, you have two publishing source options. You can either choose a branch or a custom GitHub Actions workflow. If you choose a branch, the build artifacts will be uploaded to the `publish_branch` you pass as an input to the action, which defaults to `gh-pages`. If you choose a custom GitHub Actions workflow, you need to mention that in the input `publishing_source` to the action and no artifacts/branch will be created. Please find more information on configuring a publishing source for github pages site in the [official docs](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site).
```

    # Default: '.'
    project_path: ''
    
    # Your consent to participate in Kedro-Telemetry.
    # Default: true
    telemetry_consent: ''

    # The python version used for the deployment. This should be `>=3.9`
    # Default: '3.11'
    python_version: ''

    # The publishing source for GitHub pages. This can be either 
    # 'branch' or 'workflow' based on your GitHub Pages configuration
    # Default: 'branch'
    publishing_source: ''

    # The GitHub pages publish branch to upload the artifacts 
    # if your publishing_source is a branch
    # Default: 'gh-pages'
    publish_branch: ''

    # The commit message for the deployment, if your publishing_source is a branch.
    # Defaults to your original commit message.
    # Default: ${{ github.event.head_commit.message }}
    commit_message: ''

    # The git config user.name or the owner of the commit.
    # if your publishing_source is a branch.
    # Default: 'github-actions[bot]'
    user_name: ''

    # The git config user.email or the email of the commit owner.
    # if your publishing_source is a branch.
    # Default: 'github-actions[bot]@users.noreply.github.com'
    user_email: ''

> ```

Author: Hi @noklam ,

I have mentioned that in `Configure the action` section. Also the inputs are not required and that is the reason I mentioned the default values. I tried something similar to the [checkout](https://github.com/actions/checkout/blob/main/README.md) action doc.

# Kedro-viz-share

![Kedro Logo Banner - Light](https://raw.githubusercontent.com/kedro-org/kedro/main/.github/demo-dark.png#gh-dark-mode-only)
![Kedro Logo Banner - Dark](https://raw.githubusercontent.com/kedro-org/kedro/main/.github/demo-light.png#gh-light-mode-only)

<br />
<p align="center">

![Kedro-Viz Pipeline Visualisation](https://raw.githubusercontent.com/kedro-org/kedro-viz/main/.github/img/banner.png)

</p>

<p align="center">
✨ <em> Data Science Pipelines. Beautifully Designed</em> ✨
<br />
Live Demo: <a href="https://demo.kedro.org/" target="_blank">https://demo.kedro.org/</a>
</p>

<br />

## Overview

> A GitHub action which deploys your Kedro-Viz static site on GitHub pages. This action helps in the automation of a deployment startegy mentioned in [platform agnostic sharing with Kedro-Viz](https://docs.kedro.org/projects/kedro-viz/en/v8.0.1/platform_agnostic_sharing_with_kedro_viz.html#static-website-hosting-platforms-such-as-github-pages)

Reviewer 2: ```suggestion
Kedro-viz-share is a GitHub action that simplifies the process of deploying a Kedro-viz, which is a visual representation of your Kedro project, directly within the Git repository where your Kedro project is stored. By using this action, you can effortlessly showcase your Kedro-viz on GitPages. 

This action helps in the automation of a deployment strategy mentioned in [platform agnostic sharing with Kedro-Viz](https://docs.kedro.org/projects/kedro-viz/en/v8.0.1/platform_agnostic_sharing_with_kedro_viz.html#static-website-hosting-platforms-such-as-github-pages)
```

## Prerequisites

> - **GitHub Pages Setup:** Configure your repository for [GitHub Pages](https://docs.github.com/en/pages/quickstart).

Reviewer 2: should we have one more pre-requisite 
- Create a Github repository with your Kedro project

                    - name: Set up Python
                      uses: actions/setup-python@v5
                      with:
                        # Requires python >= 3.9
                        python-version: 3.11 
                      # This installs the Kedro-project dependencies
                    - name: Install Project Dependencies
                      run: |
                        python -m pip install --upgrade pip
                        # This is not required if your Kedro Project 
                        # is at the root of your GitHub Repository
                        cd demo-project 
                        pip install -r requirements.txt
                      # Using the action
                    - name: Deploy Kedro-Viz to GH Pages 
                      uses: kedro-org/kedro-viz-share@v1
                      with:
                        # This is not required if your Kedro Project 
                        # is at the root of your GitHub Repository
                        project_path: 'demo-project'     
    ```

## Test the action

After you've completed the configuration, trigger the workflow as per the workflow trigger condition.

> - The GitHub workflow `Publish and share Kedro Viz` should start. You can find it in the Actions tab with your commit message.

Reviewer 2: ```suggestion
- Once triggered, the GitHub workflow "Publish and share Kedro Viz" will automatically start and can be found in the Actions tab with your commit message.
```

Reviewer 2: [nit]

> - Once the workflow is successfully completed, the artifacts of your Kedro-Viz static site will be uploaded to the `publish-branch` input specified to the action if your GitHub pages publishing source is branch.

Reviewer 2: [nit] 
```suggestion
- If the GitHub Pages publishing source is a branch, then the artifacts of your Kedro-Viz static site will be uploaded to the publish-branch input specified in the action upon successful completion of the workflow.
```

- If your GitHub pages publishing source is a custom GitHub Actions workflow, you need to select the above workflow `.github/workflows/kedro-viz-share.yml` as the source.
- You can access the static site at `http://<username>.github.io/<repo-name>`

## Credits

> The list of third party actions used in this project, with due credits to their authors and license terms. More details can be found inside the folder of each action.

Reviewer 2: would it makes sense to list them here?
